services/net/syndication/SyndicationAction.cs

```diff
@@ -86,20 +86,24 @@ private async Task ImportFeedAsync(IIngestServiceActionManager manager, Syndicat
                 var link = item.Links.FirstOrDefault(l => l.RelationshipType == "alternate")?.Uri;
                 if (String.IsNullOrWhiteSpace(item.Id)) item.Id = link?.ToString() ?? throw new InvalidOperationException("Feed item requires a valid 'Id' or 'Link'.");
 
+                var sourceContent = CreateSourceContent(manager.Ingest, item);
+
                 // Fetch content reference.
-                var reference = await this.FindContentReferenceAsync(manager.Ingest.Source?.Code, item.Id);
+                var reference = await this.FindContentReferenceAsync(manager.Ingest.Source?.Code, sourceContent.Uid);
                 if (reference == null)
                 {
-                    reference = await AddContentReferenceAsync(manager.Ingest, item);
+                    reference = await AddContentReferenceAsync(manager.Ingest, item, sourceContent);
+
                     await FetchContent(manager.Ingest, item, link);
                 }
-                else if ((reference.Status == (int)WorkflowStatus.InProgress && reference.UpdatedOn?.AddHours(1) < DateTime.UtcNow) ||
+                else if (reference.Status != (int)WorkflowStatus.Imported &&
```

Comment:
If I'm reading this correctly it would avoid all `Imported` references.  This would stop the logic of the second part of this statement related to content stuck in `Received`.  Not certain this is desirable.

Reply:
Yah this is the complicated condition I was talking about. The idea is to not touch content that is already in the Imported status.  But it has so many other conditions that I feel it's a bit tangled.  Might be more readable if it was broken into a series of else-if for each condition maybe?

Reply:
Up to you, but your change would break the existing behaviour.